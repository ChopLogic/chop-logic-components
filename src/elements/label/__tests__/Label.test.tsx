import { ChopLogicIcon } from '@elements';
import { ChopLogicIconName } from '@enums';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import ChopLogicLabel from '../Label';

describe('ChopLogicLabel', () => {
  const testProps = {
    label: 'Test label',
    required: true,
    inputId: 'test-input-id',
  };

  it('should render correctly as required label', () => {
    const { asFragment } = render(<ChopLogicLabel {...testProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly as not required label', () => {
    const { asFragment } = render(<ChopLogicLabel {...testProps} required={false} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly without text', () => {
    const { asFragment } = render(<ChopLogicLabel {...testProps} required={false} isTextHidden />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render left icon', () => {
    const { asFragment } = render(
      <ChopLogicLabel {...testProps} required={false} icon={<ChopLogicIcon name={ChopLogicIconName.Back} />} iconPosition='left' />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render right icon', () => {
    const { asFragment } = render(
      <ChopLogicLabel {...testProps} required={false} icon={<ChopLogicIcon name={ChopLogicIconName.Forward} />} iconPosition='left' />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
