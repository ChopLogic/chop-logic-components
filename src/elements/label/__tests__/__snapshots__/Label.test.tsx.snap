// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChopLogicLabel > should render correctly as not required label 1`] = `
<DocumentFragment>
  <label
    class="sc-egkSDF nXWIS"
    for="test-input-id"
  >
    <span>
      Test label
    </span>
  </label>
</DocumentFragment>
`;

exports[`ChopLogicLabel > should render correctly as required label 1`] = `
<DocumentFragment>
  <label
    class="sc-egkSDF nXWIS"
    for="test-input-id"
  >
    <span>
      Test label
      <abbr
        title="required"
      >
        *
      </abbr>
    </span>
  </label>
</DocumentFragment>
`;

exports[`ChopLogicLabel > should render correctly without text 1`] = `
<DocumentFragment>
  <label
    class="sc-egkSDF nXWIS"
    for="test-input-id"
  />
</DocumentFragment>
`;

exports[`ChopLogicLabel > should render left icon 1`] = `
<DocumentFragment>
  <label
    class="sc-egkSDF nXWIS"
    for="test-input-id"
  >
    <span
      aria-label="SVG Icon Fallback"
      style="font-size: 1rem;"
    >
      ⬜
    </span>
    <span>
      Test label
    </span>
  </label>
</DocumentFragment>
`;

exports[`ChopLogicLabel > should render right icon 1`] = `
<DocumentFragment>
  <label
    class="sc-egkSDF nXWIS"
    for="test-input-id"
  >
    <span
      aria-label="SVG Icon Fallback"
      style="font-size: 1rem;"
    >
      ⬜
    </span>
    <span>
      Test label
    </span>
  </label>
</DocumentFragment>
`;
